### PR TITLE
feat(api): add ovhcloud api call, the escape hatch out of partial coverage

### PR DESCRIPTION
**Pull request by Denis-hamon**
## The problem

The CLI covers **19 of the 126** dedicated-server API paths, and **8 of its 62** write operations. For everything else, a customer has to sign requests by hand — timestamp, application key, consumer key, method, URL and body, hashed — or fall back to an SDK. In practice they fall back to the Manager, which is the outcome a CLI exists to avoid.

## The escape hatch already existed, and nothing said so

It shipped as `webhosting api call`. Its generated help reads *"Call any **webhosting** API path"* — but the implementation never restricted the path to `/hosting/web`. The only path handling adds a leading slash and a version prefix. So today, on `main`:

```console
$ ovhcloud webhosting api call GET /dedicated/server
["ns3168421.ip-51-77-12.eu", …]
```

A working, general escape hatch, filed under a product it has nothing to do with and documented as something narrower than it is. I verified this with a test rather than by reading, because it seemed too odd to be true.

## What this PR does

`ovhcloud api call <method> <path>`, at the root, named for what it does. The implementation moves to `internal/services/apicall`; `webhosting api call` delegates to it, so anyone who scripted the old spelling keeps working — there is a test for that.

Three fixes come with the move:

| | Before | After |
|---|---|---|
| `/v2` paths | `/v2/publicCloud/project` → `/v1/v2/publicCloud/project` | left untouched |
| Unsupported verb | `PATCH` sent, refused by the API | named locally, nothing sent |
| Preview | none | `--dry-run` prints the request instead of sending it |

```console
$ ovhcloud api call DELETE /dedicated/server/nsXXXX/secondaryDnsDomains/example.com --dry-run
🔍 Dry run: nothing was sent. This would have been called:
  DELETE /v1/dedicated/server/nsXXXX/secondaryDnsDomains/example.com
```

## On safety

This is a raw passthrough: none of the confirmations or parameter validation the product commands carry apply to it. That is the nature of an escape hatch, and the help says so in as many words. `--dry-run` is the guard, which is why it is in this PR rather than a later one.

If you would rather it refused writes altogether, or required an explicit flag for them, say so — it is a small change and your call, not mine.

## Tests

Six, three of them with a positive control:

- removing the `return` after the dry-run report → the request leaves, `nothing must leave` fails
- dropping the `/v2` exemption → `normalizePath("/v2/publicCloud/project")` fails
- disabling the method check → `PATCH` reaches the API, `RejectsAnUnsupportedMethod` fails

## Documentation

Adds two pages, **modifies none** — no global flag was touched, so this does not trigger the 911-page regeneration that put #231 beyond Copilot's reach.

Refs: BareMetal CLI audit, requirement C1 (LVL2-16355)

### CODE REVIEW

File: internal/cmd/api.go [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  ovhcloud api call GET /v2/publicCloud/project
+
+  # Send a body, checking first what would leave
+  ovhcloud api call PUT /dedicated/server/ns3168421.ip-51-77-12.eu --from-file body.json --dry-run`,
+		Args: cobra.ExactArgs(2),
+		Run:  apicall.Call,
+	}
+
+	addParameterFileFlags(apiCallCmd, true, nil, "", "", "", nil)
+	addInteractiveEditorFlag(apiCallCmd)
```
[the remark this answers is not in the document]
Comment (Denis-hamon) [reply internal/cmd/api.go:58]: Fixed: `markFlagsMutuallyExclusive(apiCallCmd, "from-file", "editor")`.

File: internal/cmd/api.go [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+	apiCallCmd.Flags().BoolVar(&apicall.DryRun, "dry-run", false,
+		"Print the request that would be sent, without sending it")
+
+	apiCmd.AddCommand(apiCallCmd)
+	rootCmd.AddCommand(apiCmd)
```
[the remark this answers is not in the document]
Comment (Denis-hamon) [reply internal/cmd/api.go:67]: Fixed: the index lists `ovhcloud api`.

File: internal/services/apicall/apicall.go [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+	}
+
+	// A 204 leaves nothing to render, and printing an empty object would read
+	// as "the API answered nothing" rather than "it answered nothing to say".
+	if result == nil {
+		display.OutputInfo(&flags.OutputFormatConfig, nil, "✅ %s %s completed", method, path)
+		return
+	}
+
+	display.OutputWithFormat(&display.OutputMessage{Details: result}, &flags.OutputFormatConfig)
```
[the remark this answers is not in the document]
Comment (Denis-hamon) [reply internal/services/apicall/apicall.go]: Fixed: the raw value goes through `display.OutputRaw`, which defaults to JSON
and evaluates custom formats against the value itself. The one place that still
uses the envelope is `webhosting api call`, which published `{"details": …}` on
main before this branch — kept on purpose so existing `jq` expressions do not
break.

File: internal/services/apicall/apicall.go [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+
+	switch {
+	case flags.ParametersFile != "":
+		content, err := readFile(flags.ParametersFile)
+		if err != nil {
+			return nil, fmt.Errorf("failed to read parameters file: %w", err)
+		}
+		if err := json.Unmarshal(content, &body); err != nil {
+			return nil, fmt.Errorf("failed to parse parameters file: %w", err)
+		}
```
[the remark this answers is not in the document]
Comment (Denis-hamon) [reply internal/services/apicall/apicall.go]: Fixed: the validated bytes are kept as `json.RawMessage`, so an integer above
2^53 is sent unchanged.

**Comment by Denis-hamon** [comment]
A defect found while re-running the review sandbox against the reassembled stack, and fixed here in `f7b9828`.

An empty table printed its own titles wall to wall:

```
┌──────────────┬────┬────────┬───────────┬─────────┐
│subscriptionId│kind│streamId│serviceName│createdAt│
├──────────────┼────┼────────┼───────────┼─────────┤
└──────────────┴────┴────────┴───────────┴─────────┘
```

Only the data cells carried `Padding(0, 1)`, so a column came out as wide as `max(len(header), widest cell + 2)`. The header therefore only *looked* padded while some row was wider than it — and with no rows at all the padding vanished entirely. An empty list is the ordinary answer of half the `list` commands on a fresh account, not an edge case.

The header now uses the same `cellStyle`, in both tables of `display.go`. Two tests cover the two directions: an empty table pads its headers, a populated one does not double-pad. Positive control done — restoring the old style makes the first test fail on exactly that string. Nothing in the suite looked at an empty table, which is why this survived every green run.
